core/runtime/src/main/java/io/quarkus/runtime/logging/AsyncConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.quarkus.runtime.logging;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class AsyncConfig {
+
+    /**
+     * Indicates whether to log asynchronously
+     */
+    @ConfigItem(name = ConfigItem.PARENT)
+    boolean enable;
+    /**
+     * The queue length to use before flushing writing
+     */
+    @ConfigItem(defaultValue = "512")
+    int queueLength;
+
+    /**
+     * Determine whether to block the publisher (rather than drop the message) when the queue is full
+     */
+    @ConfigItem(defaultValue = "BLOCK")
+    String overflow;
```

Comment:
Please see @dmlloyd comment in regards to using `OverflowAction` instead of a `String` (I was not aware of this feature) so this can be changed to:

```java
@ConfigItem(defaultValue = "BLOCK")
OverflowAction overflow;
```

And make use in `LoggingSetupTemplate` even simpler, where we can have:

```java
asyncHandler.setOverflowAction(config.console.async.overflow);
```

Instead of:

```java
asyncHandler.setOverflowAction(OverflowAction.valueOf(config.console.async.overflow));
```

Reply:
The only thing I don't like about this approach is that it's implicitly case-sensitive.  But the enum could be enhanced later with a static `of` method which would accept strings in a case-insensitive manner (this would take precedence over the `valueOf` method which all enums carry).

Reply:
We have the same problem with log `Level` which is case-sensitive, it's widely used and more error prone.

Reply:
Instead of adding `of` to every enum class, we could create a generic method.

Reply:
Levels are different because they're inherently case-sensitive and match between what you configure and what you see, and it's an open ended
extensible set which might present a hazard when it comes to applying a blanket policy.

We could apply a blanket converter for enums that maps in a case-insensitive manner and translates from `-` to `_`.  That should be a separate issue though.

Reply:
@dmlloyd, should i keep the current implementation or revert changes  to boolean version ?

Reply:
Keep the current implementation.